### DISCUSSION TITLE
How install emgu-cv 4.5.3 with git open source

### DISCUSSION THREAD

**Discussion by md3848** [emgucv/emgucv#611] [Q&A]
EmguCV 4.5.3 — CMake 3.22 — Visual Studio 2019 16.8.3 — .NET5 — WPF

hi,! i download opencv source file from [ [this link](https://github.com/opencv/opencv) ], and compile/build with CMake, It worked perfectly and no errors occurred.

but!, when i try do this work with emgucv source file [ [this link](https://github.com/emgucv/emgucv) ] CMake show error :

![00ef99e29bcb91736ef509684358fc50959c5b75](https://user-images.githubusercontent.com/17591943/137688885-23081a8e-4e37-4082-a368-452eaf748ba5.png)

i want use this class “Emgu.CV.XPhoto.TonemapDurand” but i cant, visual sutido say :

> OpenCV: This algorithm is patented and is excluded in this configuration; Set OPENCV_ENABLE_NONFREE CMake option and rebuild the library

That’s why I try to compile/build emgu-cv myself. [ with CMake ]

in emgu-cv website [ [this link](https://emgu.com/wiki/index.php/Download_And_Installation) ], I proceeded according to the instructions, but nothing happens and an error occurs, I do not know if it is a mistake from me or …

before this i download emgu-cv from NuGet-Packages and work correctly [ .NET5 ].

**Comment by emgucv** [comment]
You will need to look into / post the files as indicated in the screen shot about to find out what went wrong with CMake project generation:
```
F:/Windows/Programing/OpenCV/S.W/Emgu.CV/4.5.3/Without Cuda/build/CmakeFiles/CmakeOutput.log
F:/Windows/Programing/OpenCV/S.W/Emgu.CV/4.5.3/Without Cuda/build/CmakeFiles/CmakeError.log
```

**Comment by md3848** [comment]
can u say when i must run this file "Build_Binary_x86-64_doc.bat" and what happen after run this file?

I do this :
1) download EmguCv from  : https://github.com/emgucv/emgucv
2) download all files like this [ below images ] and copy to EmguCv folder.

![1](https://user-images.githubusercontent.com/17591943/138423018-b97e51bb-4def-4271-93ed-51e7d447c424.PNG)

![2](https://user-images.githubusercontent.com/17591943/138423026-67607bd5-cf30-48bc-a50a-60a3dd2e2b42.PNG)

3) run CMake, set source and build folder then Press Generate.

4) after first generate, i change this flags : 

flag1 name : OPENCV_EXTRA_MODULES_PATH
flag1 value  : F:\Windows\Programing\OpenCV\S.W\EmguCV\4.5.3\emgucv\opencv_contrib\modules
/---
flag2 name : OPENCV_ENABLE_NONFREE
flag2 value  : Enable it!
//-----------------------------------------------------------------------------------------------------------------------
flag1 name : EIGEN_INCLUDE_PATH
flag1 value : F:/Windows/Programing/OpenCV/S.W/EmguCV/4.5.3/emgucv/eigen
//---
flag2 name : WITH_EIGEN
flag1 value : Enable it!

5) problems : how set VTK and FREETYPE and .... i set path but not accept, for example : 

flag1 name : VTK_DIR
flag1 value1 : F:\Windows\Programing\OpenCV\S.W\EmguCV\4.5.3\emgucv\vtk\CMake
flag1 value2 : F:\Windows\Programing\OpenCV\S.W\EmguCV\4.5.3\emgucv\vtk

i must compile VTK then set vtk-build path?
